mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessage.java

```diff
@@ -68,12 +68,12 @@ public static IndexableMessage from(MailboxMessage message, List<User> users, Te
     }
 
     private void setFlattenedAttachments(MimePart parsingResult, IndexAttachments indexAttachments) {
-        if (indexAttachments.equals(IndexAttachments.YES)) {
-            attachments = parsingResult.getAttachmentsStream()
-                    .collect(Collectors.toList());
-        } else {
-            attachments = ImmutableList.of();
-        }
+        List<MimePart> mimeparts = parsingResult.getAttachmentsStream()
+                .collect(Collectors.toList());
+
+        hasAttachment = mimeparts.stream().anyMatch(IndexableMessage::isAttachmentMimepart);
```

Comment:
I prefer having hasAttachment as a field : denormalization, you know... ES will search this more easily, and the actual ES query won't have to do a filter / find any for every messages ;-)